### PR TITLE
Proposal

### PR DESCRIPTION
**Pull request by efecarranza**
# Changelog

Add proposal to update WETH interest rates per: https://governance.aave.com/t/arc-eth-interest-rate-curve-update/10580
New interest rate strategy: https://etherscan.io/address/0x4b8D3277d49E114C8F2D6E0B2eD310e29226fe16

### CODE REVIEW

File: src/ProposalPayload.sol [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
-pragma solidity ^0.8.15;
+pragma solidity 0.8.17;
 
-import {IAaveEcosystemReserveController} from "./external/aave/IAaveEcosystemReserveController.sol";
 import {AaveV2Ethereum} from "@aave-address-book/AaveV2Ethereum.sol";
 
+interface IProposalGenericExecutor {
+    function execute() external;
+}
+
```
Comment (0xrajath) [inline src/ProposalPayload.sol]: This is not really necessary (i.e the `IProposalGenericExecutor`)

File: src/ProposalPayload.sol [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
  * @author Llama
- * @notice <DESCRIPTION>
- * Governance Forum Post:
- * Snapshot:
+ * @notice Amend ETH interest rate parameters on the Aave Ethereum v2 liquidity pool.
+ * Governance Forum Post: https://governance.aave.com/t/arc-eth-interest-rate-curve-update/10580
+ * Snapshot: https://snapshot.org/#/aave.eth/proposal/0xc78b467a96d72ea7af7d2f0ffaa4fb9e66a86f457c3f6a39a9936e6c52be1741
  */
-contract ProposalPayload {
+contract ProposalPayload is IProposalGenericExecutor {
```
Comment (0xrajath) [inline src/ProposalPayload.sol]: `IProposalGenericExecutor` is not really necessary.

File: src/ProposalPayload.sol [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
+    address public constant INTEREST_RATE_STRATEGY = 0x2Cbf7856f51660Aae066afAbaBf9C854FA6BD11f;
```
Comment (0xrajath) [inline src/ProposalPayload.sol]: The new Interest Rate Strategy seems to be modifying the `EXCESS_UTILIZATION_RATE` (from `300000000000000000000000000` to `200000000000000000000000000`) and `getMaxVariableBorrowRate`  (from `1030000000000000000000000000` to `857500000000000000000000000`) even though those parameters are not specified in the forum.

Reference:
1. New Interest Rate Contract: https://etherscan.io/address/0x2Cbf7856f51660Aae066afAbaBf9C854FA6BD11f#readContract
2. Old Interest Rate Contract: https://etherscan.io/address/0xEc368D82cb2ad9fc5EfAF823B115A622b52bcD5F#readContract
3. Governance Forum: https://governance.aave.com/t/arc-eth-interest-rate-curve-update/10580#specification-4

File: src/test/utils/AaveV2Helpers.sol [hunk 4]
```diff
@@ -0,0 +1,641 @@
+// SPDX-License-Identifier: MIT
```
Comment (0xrajath) [inline src/test/utils/AaveV2Helpers.sol:1]: Is this file not there in the `aave-address-book` lib? If not, feel free to raise a PR into the Aave governance forge template including this file. I think it'll be helpful to have it there by default.

File: src/test/ProposalPayloadE2E.t.sol [hunk 5]
```diff
@@ -1,41 +1,83 @@
...
+        // 1. Deploy new payload
+        payload = new ProposalPayload();
 
-        // Create Proposal
-        vm.prank(AAVE_WHALE);
-        proposalId = DeployMainnetProposal._deployMainnetProposal(
-            address(proposalPayload),
-            0x344d3181f08b3186228b93bac0005a3a961238164b8b06cbb5f0428a9180b8a7 // TODO: Replace with actual IPFS Hash
+        // 2. create proposal
+        vm.startPrank(GovHelpers.AAVE_WHALE);
```
Comment (0xrajath) [inline src/test/ProposalPayloadE2E.t.sol]: I didn't realise the Gov Helpers natively had an AAVE WHALE constant. Feel free to raise a PR in the Aave governance forge template updating this as well. I think it'll be a good and helpful improvement.

File: src/test/ProposalPayloadE2E.t.sol [hunk 6]
```diff
@@ -1,41 +1,83 @@
...
+            isFrozen: false
+        });
+
+        AaveV2Helpers._validateReserveConfig(expectedConfig, allConfigsAfter);
+
+        AaveV2Helpers._validateInterestRateStrategy(
+            WETH,
+            ProposalPayload(payload).INTEREST_RATE_STRATEGY(),
+            InterestStrategyValues({
+                excessUtilization: 20 * (AaveV2Helpers.RAY / 100),
```
Comment (0xrajath) [inline src/test/ProposalPayloadE2E.t.sol:64]: Are you sure this was the old interest rate strategy's excess utilization value?

File: src/test/ProposalPayloadE2E.t.sol [hunk 7]
```diff
@@ -1,41 +1,83 @@
...
+            isActive: true,
+            isFrozen: false
+        });
+
+        AaveV2Helpers._validateReserveConfig(expectedConfig, allConfigsAfter);
+
+        AaveV2Helpers._validateInterestRateStrategy(
+            WETH,
+            ProposalPayload(payload).INTEREST_RATE_STRATEGY(),
+            InterestStrategyValues({
```
Comment (0xrajath) [inline src/test/ProposalPayloadE2E.t.sol:63]: Can you also check for the max variable borrow rate parameter in this?

**Comment by 0xrajath** [comment]
@efecarranza Can you also update the README to include a description and specification of what's happening in this repo?
You can use this as reference: https://github.com/llama-community/aave-bal-bonding-curve#readme

File: src/ProposalPayload.sol (hunk 3 above)
Comment (efecarranza) [reply src/ProposalPayload.sol]: @0xrajath hey, those two are set automatically on the constructor: 

https://github.com/llama-community/aave-interest-rate-strategy-deployer/blob/main/src/DefaultReserveInterestRateStrategy.sol#L66

while the `getMaxVariableBorrowRate` depends on the variable slope:

https://github.com/llama-community/aave-interest-rate-strategy-deployer/blob/main/src/DefaultReserveInterestRateStrategy.sol#L96

File: src/test/ProposalPayloadE2E.t.sol (hunk 6 above)
Comment (efecarranza) [reply src/test/ProposalPayloadE2E.t.sol:64]: This is the new excess utilization value.

File: src/test/ProposalPayloadE2E.t.sol (hunk 7 above)
Comment (efecarranza) [reply src/test/ProposalPayloadE2E.t.sol:63]: MaxVariableBorrowRate is not part of the struct unfortunately, I can check separately for what I expect it to be.

File: src/ProposalPayload.sol (hunk 3 above)
Comment (0xrajath) [reply src/ProposalPayload.sol]: Oh cool. I didn't realize that. Makes sense!

**Review by 0xrajath — APPROVED** [review]
LGTM

File: README.md [hunk 8]
```diff
@@ -48,7 +66,7 @@ Duplicate `.env.example` and rename to `.env`:
 
 ### Deploy and Verify
 
-- `make deploy-payload` - deploy and verify payload on mainnet
+- `make deploy-contracts` - deploy and verify contracts on mainnet
```
Comment (sendra) [inline README.md]: Missing this command on Makefile

File: README.md [hunk 9]
```diff
@@ -1,35 +1,53 @@
...
+## Specification
 
-It requires [Foundry](https://github.com/gakonst/foundry) installed to run. You can find instructions here [Foundry installation](https://github.com/gakonst/foundry#installation).
+The Proposal Payload does the following:
 
-### GitHub template
+1. Sets the new interest rate strategy address.
 
-It's easiest to start a new project by clicking the ["Use this template"](https://github.com/llama-community/aave-governance-forge-template).
+The new interest rate strategy is deployed here: https://etherscan.io/address/0x2cbf7856f51660aae066afababf9c854fa6bd11f
```
Comment (sendra) [inline README.md:40]: Would also add the new rate strategy parameters, or even better the table with the old vs new values

**Review by sendra — APPROVED** [review]

**Pull request merged**